segmentstore/server/src/main/java/io/pravega/segmentstore/server/containers/StreamSegmentMetadata.java

```diff
@@ -353,7 +353,11 @@ public synchronized boolean isActive() {
 
     @Override
     public synchronized SegmentProperties getSnapshot() {
-        return StreamSegmentInformation.from(this).attributes(new HashMap<>(getAttributes())).build();
+        return StreamSegmentInformation.from(this)
```

Comment:
Why don't we add these things to `SegmentProperties` interface so we don't have to remember to set them everywhere? You can just go back to `StreamSegmentInformation` and update the `from` method which should automatically take effect wherever it's used.

Reply:
SegmentProperties is used in Storage interface and lots of other places. Changing the contract on that interface is a much bigger change that I don't want to do as part of this change.